### PR TITLE
refactor: Make module a V2 Go Module

### PR DESCRIPTION
**Pull request by lenny-intel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the current behavior?
Module is a V0 Go Module

## What is the new behavior?
Module is now a V2 Go Module

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

**Comment by lenny-intel** [comment]
Hold merging this PR until DevOps has added semver to this repo.

File: clients/agent/README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ The system management agent client is used by Go services or other Go code to co
 ### How To Use ###
 To use the management agent client package you first need to import the library into your project:
 ```
-import "github.com/edgexfoundry/go-mod-core-contracts/clients/agent"
+import "github.com/edgexfoundry/go-mod-core-contracts/v2/v2/clients/agent"
```
Comment (ernestojeda) [inline clients/agent/README.md:8]: Is this correct?

File: v2/clients/http/common.go [hunk 2]
```diff
@@ -8,11 +8,11 @@ package http
...
-	"github.com/edgexfoundry/go-mod-core-contracts/errors"
-	"github.com/edgexfoundry/go-mod-core-contracts/v2"
-	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/http/utils"
-	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/interfaces"
-	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/clients/http/utils"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/clients/interfaces"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/dtos/common"
```
Comment (ernestojeda) [inline v2/clients/http/common.go:15]: Is this correct?

**Review by ernestojeda — CHANGES_REQUESTED** [review]
See my comments about the double `/v2/v2`

**Review by ernestojeda — DISMISSED** [review]

File: clients/agent/README.md (hunk 1 above)
Comment (lenny-intel) [reply clients/agent/README.md:8]: Yes, for now. Second `v2`v will be dropped when we remove the v1 code.

**Review by ernestojeda — APPROVED** [review]

**Pull request merged**